src/awx_plugins/interfaces/_temporary_private_api.py

```diff
@@ -4,6 +4,52 @@
 The hope is that it will be refactored into something more standardized.
 """
 
+from collections.abc import Callable
+
+
+class Credential:
+    """Input supplied by the user.
+
+    Satisfies ManagedCredentialType inputs want(s).
+    """
+
+    def __init__(self: 'Credential') -> None:
+        self.inputs: dict[str, bool | str | int | float | None] = {}
+
+    def get_input(
+            self: 'Credential',
+            field_name: str,
+            default: bool | str | int | float | None = None,
+    ) -> bool | str | int | float | None:
```

Comment:
I saw this composite type being repeated in multiple places. So I recommend recording it as `GenericOptionalPrimitiveType = bool | str | int | float | None` at the top or even in a dedicated module.

Then, you'd be able to reuse it as follows here:
```suggestion
        self.inputs: dict[str, GenericOptionalPrimitiveType] = {}

    def get_input(
            self: 'Credential',
            field_name: str,
            default: GenericOptionalPrimitiveType = None,
    ) -> GenericOptionalPrimitiveType:
```

and also in other places in this PR.